include/openssl/ssl.h.in

```diff
@@ -194,14 +194,9 @@ extern "C" {
  * DEPRECATED IN 3.0.0, in favor of OSSL_default_ciphersuites()
  * Update both macro and function simultaneously
  */
-#  if !defined(OPENSSL_NO_CHACHA) && !defined(OPENSSL_NO_POLY1305)
-#   define TLS_DEFAULT_CIPHERSUITES "TLS_AES_256_GCM_SHA384:" \
-                                    "TLS_CHACHA20_POLY1305_SHA256:" \
-                                    "TLS_AES_128_GCM_SHA256"
-#  else
-#   define TLS_DEFAULT_CIPHERSUITES "TLS_AES_256_GCM_SHA384:" \
+#  define TLS_DEFAULT_CIPHERSUITES "TLS_AES_256_GCM_SHA384:" \
+                                   "TLS_CHACHA20_POLY1305_SHA256:" \
                                    "TLS_AES_128_GCM_SHA256"
-#  endif
 # endif
```

Comment:
I have some doubts it will work.

For TLS versions <= 1.2 because there was some runtime detection of the ciphers necessary for a particular ciphersuite. I didn't find such a check in this code so it means a client will announce the TLS_CHACHA20_POLY1305_SHA256 ciphersuite even without any ChaCha support and server may choose it.

Reply:
> I didn't find such a check in this code

I didn't add a new check in this PR because one already exists in master:

https://github.com/openssl/openssl/blob/a2a5506b9329b978a2a5b11a518b9789446ad310/ssl/ssl_ciph.c#L327-L337

I just tested it, and can confirm that compiling this PR with "no-chacha" there are no CHACHA based ciphersuites offered by the client

Reply:
Got it. Thanks!